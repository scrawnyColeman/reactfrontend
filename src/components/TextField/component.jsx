import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { default as MaterialTextField } from '@material-ui/core/TextField';
import { materialStyles, textFieldOverrideStyles } from '../../constants/styles';

const TextField = ({ defaultValue, onChange, disabled = false, multiline = false, rows = 1, placeholder = '' }) => {
    const useStyles = makeStyles(materialStyles);
    const classes = useStyles();

    return (
        <MaterialTextField
            className={classes.textfield}
            defaultValue={defaultValue}
            id="standard-multiline-flexible"
            multiline={multiline}
            rowsMax={rows}
            rows={rows}
            onChange={onChange}
            placeholder={placeholder}
            style={textFieldOverrideStyles}
            disabled={disabled}
        />
    );
};

export default TextField;
