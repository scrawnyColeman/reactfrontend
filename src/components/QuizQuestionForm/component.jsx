import React from 'react';
import styled from 'styled-components';
import TextField from '../TextField/component';

const Counter = styled.div`
    width: 64px;
    height: 64px;
    border-radius: 5px;
    font-size: 1.25rem;
`;
const StyledCounter = styled.div`
    padding-top: 25%;
`;
const StyledAnswer = styled.div`
    padding-top: 16px;
`;
const QuizQuestionForm = ({ questionNumber }) => {
    return (
        <>
            <Counter>
                <StyledCounter>{questionNumber}/10</StyledCounter>
            </Counter>
            <TextField placeholder={`Question ${questionNumber}`} />
            <StyledAnswer>
                <TextField placeholder={`Correct Answer:`} />
                <TextField placeholder={`Response Message:`} />
            </StyledAnswer>
            <StyledAnswer>
                <TextField placeholder={`Incorrect Answer 1:`} />
                <TextField placeholder={`Response Message 1:`} />
            </StyledAnswer>
            <StyledAnswer>
                <TextField placeholder={`Incorrect Answer 2:`} />
                <TextField placeholder={`Response Message 2:`} />
            </StyledAnswer>
        </>
    );
};
export default QuizQuestionForm;
