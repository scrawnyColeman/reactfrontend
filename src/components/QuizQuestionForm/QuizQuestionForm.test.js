import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import QuizQuestionForm from './component';
import TextField from '../TextField/component';

Enzyme.configure({ adapter: new Adapter() });

describe('QuizQuestionForm component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<QuizQuestionForm questionNumber={8} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(TextField).length).toBe(7);
    });
});
