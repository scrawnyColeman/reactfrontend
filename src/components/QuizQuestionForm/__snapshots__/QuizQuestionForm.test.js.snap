// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizQuestionForm component component matches snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      8
      /10
    </styled.div>
  </styled.div>
  <TextField
    placeholder="Question 8"
  />
  <styled.div>
    <TextField
      placeholder="Correct Answer:"
    />
    <TextField
      placeholder="Response Message:"
    />
  </styled.div>
  <styled.div>
    <TextField
      placeholder="Incorrect Answer 1:"
    />
    <TextField
      placeholder="Response Message 1:"
    />
  </styled.div>
  <styled.div>
    <TextField
      placeholder="Incorrect Answer 2:"
    />
    <TextField
      placeholder="Response Message 2:"
    />
  </styled.div>
</Fragment>
`;
